###**Bank Note Authentication**

Data were extracted from images that were taken from genuine and forged banknote-like specimens. For digitization, an industrial camera usually used for print inspection was used. The final images have 400x 400 pixels. Due to the object lens and distance to the investigated object gray-scale pictures with a resolution of about 660 dpi were gained. Wavelet Transform tool were used to extract features from images.

###Mounting the Google drive to load our dataset file **🠗**

In [1]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


###Importing necessary librabries 

In [2]:
##Dataset Link: https://www.kaggle.com/ritesaluja/bank-note-authentication-uci-data
import numpy as np
import pandas as pd

###Reading the CSV file

In [33]:
path = "/content/drive/MyDrive/Project/BankNote_Authentication.csv"
df = pd.read_csv(path)
df

variance  skewness  curtosis  entropy  class
0      3.62160   8.66610   -2.8073 -0.44699      0
1      4.54590   8.16740   -2.4586 -1.46210      0
2      3.86600  -2.63830    1.9242  0.10645      0
3      3.45660   9.52280   -4.0112 -3.59440      0
4      0.32924  -4.45520    4.5718 -0.98880      0
...        ...       ...       ...      ...    ...
1367   0.40614   1.34920   -1.4501 -0.55949      1
1368  -1.38870  -4.87730    6.4774  0.34179      1
1369  -3.75030 -13.45860   17.5932 -2.77710      1
1370  -3.56370  -8.38270   12.3930 -1.28230      1
1371  -2.54190  -0.65804    2.6842  1.19520      1

[1372 rows x 5 columns]

Our dataset has 1372 Rows and 5 columns

Our dataset is already a pre processed dataset , so we further need not do any EDA and directly move forward to Model Building. 

###**Modeling**

In [34]:
x = df.iloc[:,:-1] #Independent features
y = df.iloc[:,-1]  #Dependent feature

In [35]:
#Train Test Split
from sklearn.model_selection import train_test_split

x_train, x_test, y_train, y_test=train_test_split(x, y, test_size=0.3, random_state=0)

In [36]:
from sklearn.ensemble import RandomForestClassifier

In [38]:
rf=RandomForestClassifier()
rf.fit(x_train, y_train)
ypred=classifier.predict(x_test)

In [39]:
from sklearn.metrics import confusion_matrix, classification_report, accuracy_score

In [40]:
print(confusion_matrix(y_test,ypred))
rf_acc = accuracy_score(y_test,ypred)
print(rf_acc)
print(classification_report(ytest,ypred))

[[229   3]
 [  1 179]]
0.9902912621359223
              precision    recall  f1-score   support

           0       1.00      0.99      0.99       232
           1       0.98      0.99      0.99       180

    accuracy                           0.99       412
   macro avg       0.99      0.99      0.99       412
weighted avg       0.99      0.99      0.99       412



Here we are getting a accuracy of **99%** .

In [42]:
import pickle
pickle_out = open("rf.pkl","wb")
pickle.dump(rf, pickle_out)
pickle_out.close()